kong/resty/dns_client/init.lua

```diff
@@ -0,0 +1,782 @@
+local cjson = require("cjson.safe")
+local utils = require("kong.resty.dns_client.utils")
+local tablex = require("pl.tablex")
+local mlcache = require("kong.resty.mlcache")
+local resolver = require("resty.dns.resolver")
+
+local now       = ngx.now
+local log       = ngx.log
+local ERR       = ngx.ERR
+local WARN      = ngx.WARN
+local DEBUG     = ngx.DEBUG
+local ALERT     = ngx.ALERT
+local timer_at  = ngx.timer.at
+
+local type          = type
+local pairs         = pairs
+local ipairs        = ipairs
+local math_min      = math.min
+local string_lower  = string.lower
+local table_insert  = table.insert
+
+local parse_hosts   = utils.parse_hosts
+local ipv6_bracket  = utils.ipv6_bracket
+local search_names  = utils.search_names
+local get_next_round_robin_answers     = utils.get_next_round_robin_answers
+local get_weighted_round_robin_answers = utils.get_weighted_round_robin_answers
+
+local req_dyn_hook_run_hooks = require("kong.dynamic_hook").run_hooks
+local cycle_aware_deep_copy  = require("kong.tools.utils").cycle_aware_deep_copy
```

Comment:
```suggestion
local cycle_aware_deep_copy  = require("kong.tools.table").cycle_aware_deep_copy
```

Reply:
removed, because we do not use options table as input parameters for improving the performance